src/puppetlabs/puppetdb/query/aggregate_event_counts.clj

```diff
@@ -1,50 +1,80 @@
 (ns puppetlabs.puppetdb.query.aggregate-event-counts
   (:require [puppetlabs.puppetdb.query.event-counts :as event-counts]
+            [clojure.string :as str]
+            [puppetlabs.puppetdb.query.events :as events]
             [puppetlabs.puppetdb.jdbc :as jdbc]
             [puppetlabs.kitchensink.core :as kitchensink]))
 
 (defn- get-aggregate-sql
   "Given the `event-count-sql`, return a SQL string that will aggregate the results."
-  [event-count-sql]
+  [event-count-sql summarize_by]
   {:pre  [(string? event-count-sql)]
    :post [(string? %)]}
-  (format "SELECT SUM(CASE WHEN successes > 0 THEN 1 ELSE 0 END) as successes,
+  (format "(SELECT '%s' as summarize_by,
+           SUM(CASE WHEN successes > 0 THEN 1 ELSE 0 END) as successes,
                   SUM(CASE WHEN failures > 0  THEN 1 ELSE 0 END) as failures,
                   SUM(CASE WHEN noops > 0 THEN 1 ELSE 0 END) as noops,
                   SUM(CASE WHEN skips > 0 THEN 1 ELSE 0 END) as skips,
                   COUNT(*) as total
-           FROM (%s) event_counts" event-count-sql))
+           FROM (%s) event_counts)" summarize_by event-count-sql))
 
-(defn query->sql
-  "Convert an aggregate-event-counts `query` and a value to `summarize_by` into a SQL string.
-  Since all inputs are forwarded to `event-counts/query->sql`, look there for proper documentation."
+(defn assemble-aggregate-sql
+  "Convert an aggregate-event-counts `query` and a value to `summarize_by`
+   into a SQL string."
   [version query [summarize_by query-options]]
```

Comment:
we assemble the [summarize_by query-options] vector below seemingly for the sole purpose of unpacking it here.